frontends/ast/simplify.cc

```diff
@@ -2341,15 +2341,12 @@ bool AstNode::simplify(bool const_fold, bool at_zero, bool in_lvalue, int stage,
 	{
 		expand_genblock(str + ".");
 
-		// if this is an autonamed block is in an always_comb
-		if (current_always && current_always->attributes.count(ID::always_comb)
-				&& str.at(0) == '$')
```

Comment:
After further investigation, the root cause is that `str.at(0) == '$'` isn't accurate for identifying autonamed blocks. This issue even affects blocks outside of functions. Rather than remove these checks entirely, I'll open a PR to fix them.